salvo/README.md

```diff
@@ -0,0 +1,73 @@
+# salvo
+
+This is a framework that abstracts executing multiple benchmarks of the Envoy Proxy using [NightHawk](https://github.com/envoyproxy/nighthawk).
+
+## Example Control Documents
+
+The control document defines the data needed to excute a benchmark.  At the moment the dockerized scavenging benchmark is the only one supported.  To run the benchmark, create a file with the following example contents:
+
+JSON Example:
+
+```json
+{
+  "remote": false,
+  "dockerizedBenchmark": true,
+  "images": {
+    "reuseNhImages": true,
+    "nighthawkBenchmarkImage": "envoyproxy/nighthawk-benchmark-dev:latest",
+    "nighthawkBinaryImage": "envoyproxy/nighthawk-dev:latest",
+    "envoyImage": "envoyproxy/envoy-dev:f61b096f6a2dd3a9c74b9a9369a6ea398dbe1f0f"
+  },
+  "environment": {
+    "v4only": true,
+    "envoyPath": "envoy",
+    "outputDir": "/home/ubuntu/nighthawk_output",
+    "testDir": "/home/ubuntu/nighthawk_tests"
+  }
+}
+```
+
+YAML Example:
+
+```yaml
+remote: false
+dockerizedBenchmark: true
+environment:
+  envoyPath: 'envoy'
+  outputDir: '/home/ubuntu/nighthawk_output'
+  testDir: '/home/ubuntu/nighthawk_tests'
+  v4only: true
+images:
+  reuseNhImages: true
+  nighthawkBenchmarkImage: 'envoyproxy/nighthawk-benchmark-dev:latest'
+  nighthawkBinaryImage: 'envoyproxy/nighthawk-dev:latest'
+  envoyImage: "envoyproxy/envoy-dev:f61b096f6a2dd3a9c74b9a9369a6ea398dbe1f0f"
+```
+
+In both examples, the envoy image being tested is a specific hash.  This hash can be replaced with "latest" to test the most recently created image against the previous image built from the prior Envoys master commit.
+
+
+## Building Salvo
+
+```bash
+bazel build //:salvo
+```
+
+## Running Salvo
+
+```bash
+bazel-bin/salvo --job ~/test_data/demo_jobcontrol.yaml
+```
+
+## Testing Salvo
+
+```bash
+ bazel test //test:*
```

Comment:
Similarly,  maybe this might be good to add to `ci/do_ci.sh` to make sure these tests run without failures in CI.

Reply:
Looks like the test failures are due to dependency issues (pytest, and python3.5).   I'll look at addressing this in a subsequent PR.  We were targeting python3.6 as interpreter, but it appears that python3.5 is the newest version available in the `envoyproxy/envoy-build` container.

Reply:
Are you sure you can't access something like Python 3.8 by using `python3.8` directly? I've had to do this in a few places in Envoy CI.

Reply:
I can look into doing this. I started digging into this again today and had not ventured into Envoy to see how it's handled there.